### PR TITLE
Manual Update and minor restructure of tests

### PR DESCRIPTION
**Pull request by yanliu18**
There has been some significant changes since the last time the manual is updated.
These changes added more functionalities to `kmir` and make the tool easier to use.
Also, the instructures spreads in two README files causing some overlapping and inconsistencies.
They are now merged in to one and restructured for end-user and developper to access what they need.
There have been a number of test directories, they are slightly restrutred to shorten the names and flatten the structures a little bit.

The next step would also need to flattern the project structure as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,45 +1,246 @@
...
+```sh
+kmir prove --spec-file kmir/src/tests/integration/test-data/prove-rs/simple-spec.k
+```
+
+#### Display KCFG
+```sh
+kmir show-kcfg --spec-file kmir/src/tests/integration/test-data/prove-rs/simple-spec.k --save-directory <path to folder>
+```
+
+In order to save `kcfgs` the flag `--save-directory` can be added with a path to the directory to save the kcfgs to (this dir must already be created).
```
[the remark this answers is not in the document]
Comment (dkcumming) [reply README.md:106]: Opps I think I highlighted too much, I mean the explanation on 107 is meant for `kmir prove`

File: README.md [hunk 2]
```diff
@@ -1,45 +1,246 @@
...
+options:
+  -h, --help            show this help message and exit
+  --verbose, -v         Verbose output.
+  --debug               Debug output.
+  --definition-dir DEFINITION_DIR
+                        Path to LLVM definition to use.
+  --input {program,binary,json,kast,kore}
+                        Input mode
+  --output {pretty,program,json,kore,kast,none}
+                        Output mode
```
[the remark this answers is not in the document]
Comment (yanliu18) [reply README.md:71]: I wrote out the options before, but I found the helping message more detailed and systematic.
So, I replaced the text with the helping message.

I feel like it is good to put here, giving people a clear idea of what they can have.
Once they get started, most of the introductory text won't be needed at all.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (yanliu18) [reply README.md:106]: I didn't find a line 107? I believe I copy pasted from what you've added before.
Though, I am not sure who should appear first.

Do you want a `--save-directory` tag for `kprove` too?

File: README.md [hunk 3]
```diff
@@ -1,45 +1,246 @@
...
+
+0. Pick a MIR/Rust program as a running example, or a several of them
+1. Modify K files
+2. Rebuild the K definition
+3. Testout the new definition on the running example
+4. Running integration tests
+
+However, from step 3, the following alternative steps avoids rebuilding from scratch:
+1. Run `poetry run kbuild kompile llvm` to re-kompile the semantics
+2. Test the running example with a targetted `kmir <command> <path to example.mir>`
```
[the remark this answers is not in the document]
Comment (yanliu18) [reply README.md:218]: Yes, you are write, in that case, just go with make build is the best. 
This para was meant for `kparse`.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dkcumming) [reply README.md:86]: Sorry must of been a typo, I was referring to `In order to save `kcfgs` the flag `--save-directory` can be added with a path to the directory to save the kcfgs to (this dir must already be created).` I don't think the flag needs to be added to the command as it is optional, but it is good to mention that the directory needs to be created at the stage of `kmir prove` I thought.

**Review by dkcumming — APPROVED** [review]

**Pull request merged**